tests/transformers/test_transformers_autolog.py

```diff
@@ -0,0 +1,511 @@
+        "I thoroughly dislike artisinal bathroom cleaning.",
+    ]
+    train_labels = [1, 0]
+    eval_texts = [
+        "It was an excellent experience.",
+        "I'd rather pick my teeth with a rusty pitchfork.",
+    ]
+    eval_labels = [1, 0]
+
+    train_encodings = tokenizer(train_texts, truncation=True, padding=True)
+    eval_encodings = tokenizer(eval_texts, truncation=True, padding=True)
+
+    class CustomDataset(torch.utils.data.Dataset):
+        def __init__(self, encodings, labels):
+            self.encodings = encodings
+            self.labels = labels
+
+        def __getitem__(self, idx):
+            item = {key: torch.tensor(val[idx]) for key, val in self.encodings.items()}
+            item["labels"] = torch.tensor(self.labels[idx])
+            return item
+
+        def __len__(self):
+            return len(self.labels)
+
+    train_dataset = CustomDataset(train_encodings, train_labels)
+    eval_dataset = CustomDataset(eval_encodings, eval_labels)
+
+    training_args = TrainingArguments(
+        output_dir=str(tmp_path.joinpath("results")),
+        num_train_epochs=1,
+        per_device_train_batch_size=4,
+        logging_dir=str(tmp_path.joinpath("logs")),
+        report_to="none",
+    )
+
+    def model_init():
+        return DistilBertForSequenceClassification.from_pretrained(
+            "distilbert-base-uncased", num_labels=2
+        )
+
+    trainer = Trainer(
+        model_init=model_init,
+        args=training_args,
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+    )
+
+    def my_hp_space_optuna(trial):
+        return {
+            "learning_rate": trial.suggest_float("learning_rate", 1e-5, 1e-1, log=True),
+        }
+
+    best_run = trainer.hyperparameter_search(
+        hp_space=my_hp_space_optuna,
+        backend="optuna",
+        n_trials=2,
+        direction="minimize",
+    )
+
+    best_training_args = TrainingArguments(
+        output_dir=str(tmp_path.joinpath("best_results")),
+        num_train_epochs=1,
+        per_device_train_batch_size=4,
+        learning_rate=best_run.hyperparameters["learning_rate"],
+        logging_dir=str(tmp_path.joinpath("best_logs")),
+    )
+
+    return Trainer(
+        model=model_init(),
+        args=best_training_args,
+        train_dataset=train_dataset,
+    )
+
+
+def test_setfit_does_not_autolog(setfit_trainer):
+    mlflow.autolog()
+
+    setfit_trainer.train()
+
+    last_run = mlflow.last_active_run()
+    assert not last_run
+    preds = setfit_trainer.model(
+        ["Always carry a towel!", "The hobbits are going to Isengard", "What's tatoes, precious?"]
+    )
+    assert len(preds) == 3
+
+
+def test_transformers_trainer_does_not_autolog_sklearn(transformers_trainer):
+    mlflow.autolog()
+
+    transformers_trainer.train()
+
+    last_run = mlflow.last_active_run()
+    assert last_run.data.metrics is not None
+    assert last_run.data.params is not None
```

Comment:
If we replace `mlflow.autolog()` with `mlflow.sklearn.autolog()`, would that autolog an sklearn model?

Reply:
It won't. To make it clear for this behavior, I changed the explicit override to the global autolog to enable autolog for sklearn in this test